sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsCheckpointMark.java

```diff
@@ -56,6 +64,7 @@ public void finalizeCheckpoint() {
       }
     }
     messages.clear();
+    oldestPendingTimestamp = System.currentTimeMillis();
```

Comment:
Rather than being `Instant.now()`, maybe it would make more sense to be the maximum timestamp among all of the elements in the batch? This assumes that event time is in increasing order, but it's again better than now because there might be old messages still.